internal/error.go

```diff
@@ -1029,3 +1055,16 @@ func getErrType(err error) string {
 
 	return t.Name()
 }
+
+func isBenignApplicationError(err error) bool {
+	var appError *ApplicationError
+	return errors.As(err, &appError) && appError.Category() == ApplicationErrorCategoryBenign
```

Comment:
I don't think we should assume causes no matter the depth as benign. Explained at https://github.com/temporalio/sdk-java/pull/2485#discussion_r2059154968.

```suggestion
	appError, _ := err.(*ApplicationError)
	return appError != nil && appError.Category() == ApplicationErrorCategoryBenign
```

Reply:
Yes we should only look at the top level error